streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -147,6 +147,71 @@ public String toString() {
         }
     }
 
+    /**
+     * A TopicNode is a node that contains topic information and upstream/downstream TopicsInfo. Graph built of TopicNode and TopicsInfoNode is useful
```

Comment:
Like we discussed offline, the downstream is actually not necessary.

Reply:
yeah, absolutely, updated :)